packages/core-data/src/hooks/use-entity-record.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { Status } from './constants';
+
+interface EntityRecordResolution {
+	/** The requested entity record */
+	record: Object;
+
+	/** The requested entity record with any edits applied*/
+	editedRecord: Object;
+
+	/** Is the record still being resolved? */
+	isResolving: boolean;
+
+	/** Is the record resolved by now? */
+	hasResolved: boolean;
```

Comment:
I guess we don't need this either, as this is essentially the same as `!isResolving`.

Reply:
Not exactly, `isResolving` and `hasResolved` may both be true when the record is being re-requested. It is confusing so I'll add that to the docstring.

Reply:
> `isResolving` and `hasResolved` may both be true when the record is being re-requested.

Huh, interesting! Then what's the difference between `hasResolved` and `!!data`?

Reply:
Just checked the implementation in:

https://github.com/WordPress/gutenberg/blob/b51dbd6c3ec18e45b114311764bd1a232858ef7d/packages/data/src/redux-store/metadata/selectors.js#L53-L69

> `isResolving` and `hasResolved` may both be true

I think this is not possible? Or maybe I'm missing something elsewhere 🤔 . I guess something like `isSuccess` and `isError` might be more useful than `hasResolved`. It's also arguably more approachable given that the name directly inherits the `status`.